hrqb/tasks/sql/employees.sql

```diff
@@ -40,19 +42,27 @@ appt_termination_txns as (
         ad.MIT_ID,
         ad.HR_POSITION_KEY,
         at.HR_PERSONNEL_ACTION,
-        at.HR_ACTION_REASON as TERMINATION_REASON
+        at.HR_ACTION_REASON as TERMINATION_REASON,
+        row_number() over (
+            partition by MIT_ID
+            order by APPT_BEGIN_DATE desc, APPT_END_DATE desc
+        ) as termination_txn_row_num
     from HR_APPT_ACTION_DETAIL ad
     left join HR_PERSONNEL_ACTION_TYPE at on at.HR_PERSONNEL_ACTION_TYPE_KEY = ad.HR_PERSONNEL_ACTION_TYPE_KEY
     where at.HR_PERSONNEL_ACTION in ('Retirement','Termination')
 ),
+last_appt_termination_txn as (
+    select * from appt_termination_txns
+    where termination_txn_row_num = 1
+),
```

Comment:
The addition of the `row_number()` sorts the termination reasons, and the new `last_appt_termination_txn` CTE allows for getting only the most recent one, i.e. the "last" termination reason.

Reply:
I follow! Hmm, is there every a scenario like the following?: 
```python
[{
    "APPT_BEGIN_DATE": "2024-01-01", 
    "APPT_END_DATE": "2025-01-01",
    "termination_txn_row_num":  1
   }, 
 {
   "APPT_BEGIN_DATE": "2023-12-31"
   "APPT_END_DATE": "2025-02-01"
   "termination_txn_row_num": 2
 }]
```